lib/src/main/kotlin/nodes/abstractNode.kt

```diff
@@ -0,0 +1,6 @@
+package nodes
+
+abstract class abstractNode<K: Comparable<K>, V, someNode: abstractNode<K, V, someNode>>(val key:K, var value: V) {
```

Comment:
Согласно котлиновскому [Coding conventions](https://kotlinlang.org/docs/coding-conventions.html#naming-rules) названия для классов лучше начинать с заглавной буквы.
```suggestion
abstract class AbstractNode<K: Comparable<K>, V, someNode: abstractNode<K, V, someNode>>(val key:K, var value: V) {
```